service/src/main/java/tds/config/repositories/impl/TimeLimitConfigurationRepositoryImpl.java

```diff
@@ -0,0 +1,104 @@
+package tds.config.repositories.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+import tds.config.TimeLimitConfiguration;
+import tds.config.repositories.TimeLimitConfigurationRepository;
+import tds.config.repositories.impl.mappers.TimeLimitsRowMapper;
+
+import javax.sql.DataSource;
+import java.util.Optional;
+
+@Repository
+public class TimeLimitConfigurationRepositoryImpl implements TimeLimitConfigurationRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigRepositoryImpl.class);
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public TimeLimitConfigurationRepositoryImpl(DataSource dataSource) {
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(final String clientName) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("clientName", clientName);
+
+        final String SQL =
+                "SELECT \n" +
+                "   clientname, \n" +
+                "   _efk_testid AS assessmentId, \n" +
+                "   environment, \n" +
+                "   opprestart AS examRestartWindowMinutes, \n" +
+                "   oppdelay AS examDelayDays, \n" +
+                "   interfacetimeout AS interfaceTimeoutMinutes, \n" +
+                "   requestinterfacetimeout AS requestInterfaceTimeoutMinutes, \n" +
+                "   tacheckintime AS taCheckinTimeMinutes \n" +
+                "FROM \n" +
+                "   session.timelimits \n" +
+                "WHERE \n" +
+                "   clientname = :clientName \n" +
+                "   AND _efk_testid IS NULL";
+
+        Optional<TimeLimitConfiguration> maybeTimeLimitConfig;
+        try {
+            maybeTimeLimitConfig = Optional.of(
+                    jdbcTemplate.queryForObject(
+                            SQL,
+                            parameters,
+                            new TimeLimitsRowMapper()));
+        } catch(IncorrectResultSizeDataAccessException e) {
+            LOG.debug("{} did not return results for clientName = {}", SQL, clientName);
+            maybeTimeLimitConfig = Optional.empty();
+        }
+
+        return maybeTimeLimitConfig;
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(final String clientName, final String assessmentId) {
```

Comment:
I initially only had one method that had `clientName` and `assessmentId` parameters.  I bailed on it because the overload is clearer to me.  You can get time limit configuration for client name + assessment id or you can get time limit configuration for just client name